flink-clients/src/main/java/org/apache/flink/client/program/Client.java

```diff
@@ -271,18 +272,21 @@ public JobSubmissionResult runDetached(PackagedProgram prog, int parallelism)
 		}
 		else if (prog.isUsingInteractiveMode()) {
 			LOG.info("Starting program in interactive mode");
-			ContextEnvironment.setAsContext(this, prog.getAllLibraries(), prog.getClasspaths(),
-				prog.getUserCodeClassLoader(), parallelism, false);
-
+			ContextEnvironment.ContextEnvironmentFactory factory = ContextEnvironment.setAsContext(this,
+					prog.getAllLibraries(), prog.getClasspaths(), prog.getUserCodeClassLoader(), parallelism, false);
 			// invoke here
 			try {
 				prog.invokeInteractiveModeForExecution();
+				ContextEnvironment ctx = factory.getLastEnvironment();
+				if (ctx == null) {
+					throw new InvalidProgramException("No execution environment was created.");
```

Comment:
It's not strictly necessary. This should return a `JobSubmissionResult` with null job id then.